src/js/utils/Select.jsx

```diff
@@ -128,6 +163,29 @@ class Select extends Component {
     return (value && <div className="p-1">{value.label ?? value?.name}</div>);
   }
 
+  mapOptions(values) {
+    return (_.map(values, (value) => {
+      let option = value;
+
+      if (typeof value === 'string') {
+        return { value, label: value };
+      }
+      if (value && this.props.valueKey && !option.value) {
+        option = { ...option, value: option[this.props.valueKey] };
+      }
+
+      if (value && this.props.labelKey && !option.label) {
+        option = { ...option, label: option[this.props.labelKey] };
+      }
+
+      if (option.options) {
+        option = { ...option, options: this.mapOptions(this.props.options) };
+      }
+
+      return option;
+    }));
+  }
+
```

Comment:
As i can see it's not your code, but maybe return option directly from `if` instead of assigning to option and then return.
I mean:
```js
 mapOptions(values) {
    return (_.map(values, (value) => {
      if (typeof value === 'string') {
        return { value, label: value };
      }

      if (value && this.props.valueKey && !option.value) {
        return { ...value, value: option[this.props.valueKey] };
      }

      if (value && this.props.labelKey && !option.label) {
        return { ...value, label: option[this.props.labelKey] };
      }

      if (option.options) {
        return { ...value, options: this.mapOptions(this.props.options) };
      }

      return value;
    }));
  }
```

Reply:
yes makes sense, although I would need to reorder those actions